clion/src/main/java/org/sonarlint/intellij/clion/AnalyzerConfiguration.java

```diff
@@ -175,19 +175,33 @@ static ForcedLanguage getSonarLanguage(OCLanguageKind languageKind) {
   }
 
   private boolean usingRemoteOrWslToolchain(OCResolveConfiguration configuration) {
-    final var initializedWorkspaces = CidrWorkspace.getInitializedWorkspaces(project);
-    CPPEnvironment cppEnvironment = null;
-    for (var initializedWorkspace : initializedWorkspaces) {
-      if (initializedWorkspace instanceof CMakeWorkspace cMakeWorkspace) {
-        cppEnvironment = getCMakeCppEnvironment(cMakeWorkspace, configuration);
-      } else {
-        cppEnvironment = tryReflection(initializedWorkspace);
-        if (cppEnvironment != null) {
-          break;
+    try {
+      tryIfClassAccessible();
+      final var initializedWorkspaces = CidrWorkspace.getInitializedWorkspaces(project);
+      for (var initializedWorkspace : initializedWorkspaces) {
+        if (Class.forName("com.jetbrains.cidr.cpp.cmake.workspace.CMakeWorkspace").isInstance(initializedWorkspace)) {
+          var cppEnvironment = getCMakeCppEnvironment(initializedWorkspace, configuration);
+          if (cppEnvironment != null) {
+            return cppEnvironment.getToolSet().isRemote() || cppEnvironment.getToolSet().isWSL() || cppEnvironment.getToolSet().isDocker();
+          }
+        } else {
+          var cppEnvironment = tryReflection(initializedWorkspace);
+          if (cppEnvironment != null) {
+            return cppEnvironment.getToolSet().isRemote() || cppEnvironment.getToolSet().isWSL() || cppEnvironment.getToolSet().isDocker();
+          }
         }
       }
+      SonarLintConsole.get(project).debug("Not using remote or WSL toolchain");
+      return false;
+    } catch (ClassNotFoundException e) {
+      SonarLintConsole.get(project).debug("Could not check if remote or WSL toolchain is used");
+      return false;
     }
-    return cppEnvironment != null && (cppEnvironment.getToolSet().isRemote() || cppEnvironment.getToolSet().isWSL() || cppEnvironment.getToolSet().isDocker());
+  }
+
+  private static void tryIfClassAccessible() throws ClassNotFoundException {
+    Class.forName("com.jetbrains.cidr.cpp.cmake.workspace.CMakeWorkspace");
+    Class.forName("com.jetbrains.cidr.cpp.toolchains.CPPEnvironment");
```

Comment:
Do you really need to check both? Why? If so, a comment might be useful

Reply:
On the latest version of CLion, when ReSharper is off, these 2 classes/packages got removed. So in the current state, we don't support remote/WSL in this case.

Reply:
What I mean is: wouldn't it be sufficient to check only for one of them? Why having 2 checks?